Editor/Resources/EditorWindow/GameLiftPlugin.cs

```diff
@@ -0,0 +1,204 @@
+            RefreshProfiles();
+            
+            var contentContainer = _root.Q(className: "main__content").Children().First();
+            var landingPage = new LandingPage(contentContainer);
+            _userProfilesPage = new AwsUserProfilesPage(contentContainer,this);
+
+            _tabButtons = _root.Query<Button>(className: TabButtonClassName).ToList();
+            _tabContent = _root.Query(className: TabContentClassName).ToList();
+
+            _tabButtons.ForEach(button => button.RegisterCallback<ClickEvent>(_ => { OpenTab(button.name); }));
+
+            OpenTab("Landing");
+        }
+
+        private void ApplyText()
+        {
+            var l = new ElementLocalizer(_root);
+            l.SetElementText("Landing", Strings.TabLanding);
+            l.SetElementText("Credentials", Strings.TabCredentials);
+            l.SetElementText("Anywhere", Strings.TabAnywhere);
+            l.SetElementText("EC2", Strings.TabEC2);
+            l.SetElementText("Help", Strings.TabHelp);
+        }
+
+        private void OpenTab(string tabName)
+        {
+            _tabContent.ForEach(page =>
+            {
+                if (!page.name.StartsWith(tabName))
+                {
+                    page.RemoveFromClassList(TabContentSelectedClassName);
+                }
+                else
+                {
+                    page.AddToClassList(TabContentSelectedClassName);
+                }
+            });
+
+            _tabButtons.ForEach(button =>
+            {
+                if (button.name != tabName)
+                {
+                    button.RemoveFromClassList(TabButtonSelectedClassName);
+                }
+                else
+                {
+                    button.AddToClassList(TabButtonSelectedClassName);
+                }
+            });
+        }
+        
+        public void RefreshProfiles()
+        {
+            UpdateModel.Refresh();
+            CurrentState.AllProfiles = UpdateModel.AllProlfileNames;
+        }
+        
+        public void SetupWrapper()
+        {
+            var credentials = CoreApi.RetrieveAwsCredentials(CurrentState.SelectedProfile);
+            var client = new AmazonGameLiftClient(credentials.AccessKey, credentials.SecretKey);
+            GameLiftWrapper = new AmazonGameLiftWrapper(client);
+        }
+        
+        public void OpenS3Popup(string bucketName)
+        {
+            var popup = CreateInstance<GameLiftPluginBucketPopup>();
+            popup.Init(bucketName);
+            popup.OnConfirm += BootStrapPassthrough;
+            popup.ShowModalUtility();
+        }
+
+        private void BootStrapPassthrough(string bucketName)
+        {
+            _userProfilesPage.BootstrapAccount(bucketName);
+        }
+    }
+    
+    public struct State
+    {
+        public bool SelectedBootstrapped;
+        public string[] AllProfiles;
+        public string SelectedProfile;
+        public int SelectedFleetIndex;
+    }
+}
```

Comment:
I don't see this being referenced anywhere currently. Could you please clean this from this PR and re-add it as and when required? That will help give context of usage and what the best way to incorporate this would be.

Reply:
SelectedFleetIndex Was the only unused portion of this. AwsUserProfilesPage.cs uses the rest of these extensively. E.g line 172 AwsUserProfilesPage.cs